src/sources/soundsourceflac.cpp

```diff
@@ -435,9 +435,18 @@ void SoundSourceFLAC::flacMetadata(const FLAC__StreamMetadata* metadata) {
         DEBUG_ASSERT(kBitsPerSampleDefault != bitsPerSample);
         if (kBitsPerSampleDefault == m_bitsPerSample) {
             // not set before
-            m_bitsPerSample = bitsPerSample;
-            m_sampleScaleFactor = CSAMPLE_PEAK
-                    / CSAMPLE(FLAC__int32(1) << bitsPerSample);
+            if (bitsPerSample > 0) {
+                m_bitsPerSample = bitsPerSample;
+                // Range of signed(!) sample values: [2 ^ (bitsPerSample - 1), 2 ^ (bitsPerSample - 1) - 1]
+                // See also: https://bugs.launchpad.net/mixxx/+bug/1766042
+                const auto maxAbsSampleValue = FLAC__int32(1) << (bitsPerSample - 1);
```

Comment:
The cast here and below are IMHO unnecessary. If you think they are, use static_cast.